packages/@aws-cdk/aws-eks/lib/alb-controller.ts

```diff
@@ -0,0 +1,245 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as iam from '@aws-cdk/aws-iam';
+import { Construct } from 'constructs';
+import { Cluster } from './cluster';
+import { HelmChart } from './helm-chart';
+import { ServiceAccount } from './service-account';
+
+// v2 - keep this import as a separate section to reduce merge conflict when forward merging with the v2 branch.
+// eslint-disable-next-line
+import { Construct as CoreConstruct, Duration, Names, Stack } from '@aws-cdk/core';
+
+/**
+ * Controller version.
+ *
+ * Corresponds to the image tag of 'amazon/aws-load-balancer-controller' image.
+ */
+export class AlbControllerVersion {
+
+  /**
+   * v2.0.0
+   */
+  public static readonly V2_0_0 = new AlbControllerVersion('v2.0.0', false);
+
+  /**
+   * v2.0.1
+   */
+  public static readonly V2_0_1 = new AlbControllerVersion('v2.0.1', false);
+
+  /**
+   * v2.1.0
+   */
+  public static readonly V2_1_0 = new AlbControllerVersion('v2.1.0', false);
+
+  /**
+   * v2.1.1
+   */
+  public static readonly V2_1_1 = new AlbControllerVersion('v2.1.1', false);
+
+  /**
+   * v2.1.2
+   */
+  public static readonly V2_1_2 = new AlbControllerVersion('v2.1.2', false);
+
+  /**
+   * v2.1.3
+   */
+  public static readonly V2_1_3 = new AlbControllerVersion('v2.1.3', false);
+
+  /**
+   * v2.0.0
+   */
+  public static readonly V2_2_0 = new AlbControllerVersion('v2.2.0', false);
+
+  /**
+   * v2.2.1
+   */
+  public static readonly V2_2_1 = new AlbControllerVersion('v2.2.1', false);
+
+  /**
+   * v2.2.2
+   */
+  public static readonly V2_2_2 = new AlbControllerVersion('v2.2.2', false);
+
+  /**
+   * v2.2.3
+   */
+  public static readonly V2_2_3 = new AlbControllerVersion('v2.2.3', false);
+
+  /**
+   * v2.2.4
+   */
+  public static readonly V2_2_4 = new AlbControllerVersion('v2.2.4', false);
+
+  /**
+   * v2.3.0
+   */
+  public static readonly V2_3_0 = new AlbControllerVersion('v2.3.0', false);
+
+  /**
+   * Specify a custom version.
+   *
+   * @param version The version number.
+   */
+  public static of(version: string) {
+    return new AlbControllerVersion(version, true);
+  }
+
+  private constructor(
+    /**
+     * The version string.
+     */
+    public readonly version: string,
+    /**
+     * Whether or not its a custom version.
+     */
+    public readonly custom: boolean) {}
```

Comment:
When would you need to create a custom version?

Reply:
Added docs 👍